### PR TITLE
Adding custom type to validate empty strings in entity-dir param of create commend

### PR DESCRIPTION
**Pull request by Mun1z**
params that receive this custom type will be checked if they are empty.

`$ ml-git datasets create dataset-ex --mutability=strict --category=imgs --bucket-name=test --entity-dir=`
Will cause an error and fail the command

### CODE REVIEW

**Review by lucasncabral — COMMENTED** [review]
The PR description says:
```
params that receive this custom type will have .strip() called on them and will be checked if empty.
```
The type `NotEmptyString` applied to the option does not perform the strip. The `NotEmptyString`  type will validate the received value and check if it's an empty string. Please check if the treatment performed is the desired one.

Also, add tests to validate the fix.

File: ml_git/commands/descriptor.py [hunk 1]
```diff
@@ -481,7 +481,7 @@
             '--credentials-path': {'default': None, 'help': help_msg.CREDENTIALS_PATH,
                                    'cls': OptionRequiredIf, 'required_option': ['import-url']},
             '--unzip': {'help': help_msg.UNZIP_OPTION, 'is_flag': True},
-            '--entity-dir': {'default': '', 'help': help_msg.ENTITY_DIR}
+            '--entity-dir': {'type': NotEmptyString(),'default': '', 'help': help_msg.ENTITY_DIR}
```
[the remark this answers is not in the document]
Comment (Mun1z) [reply ml_git/commands/descriptor.py]: The PR description has updated.
The default value has changes to '.', and has added a treatment for this default value on directory creation.

**Review by GustavoMaciel — CHANGES_REQUESTED** [review]
The chosen implementation was a little off from the python standards, so I left a few suggestions on how the same thing could be accomplished in more of a python way.

File: ml_git/commands/descriptor.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lucasncabral) [reply ml_git/commands/descriptor.py]: Please take a look at the comments @GustavoMaciel posted.

File: ml_git/config.py [hunk 2]
```diff
@@ -306,6 +306,10 @@ def create_workspace_tree_structure(repo_type, artifact_name, categories, storag
                                     imported_dir, mutability, entity_dir=''):
     # get root path to create directories and files
     path = get_root_path()
+
+    if (entity_dir == '.'):
+        entity_dir = ''
+
```
[the remark this answers is not in the document]
Comment (Mun1z) [reply ml_git/config.py]: Done on last commit.

File: ml_git/commands/descriptor.py [hunk 3]
```diff
@@ -481,7 +481,7 @@
             '--credentials-path': {'default': None, 'help': help_msg.CREDENTIALS_PATH,
                                    'cls': OptionRequiredIf, 'required_option': ['import-url']},
             '--unzip': {'help': help_msg.UNZIP_OPTION, 'is_flag': True},
-            '--entity-dir': {'default': '', 'help': help_msg.ENTITY_DIR}
+            '--entity-dir': {'type': NotEmptyString(),'default': '.', 'help': help_msg.ENTITY_DIR}
```
[the remark this answers is not in the document]
Comment (Mun1z) [reply ml_git/commands/descriptor.py]: Done on last commit.

File: ml_git/config.py [hunk 4]
```diff
@@ -306,6 +306,10 @@ def create_workspace_tree_structure(repo_type, artifact_name, categories, storag
                                     imported_dir, mutability, entity_dir=''):
```
[the remark this answers is not in the document]
Comment (Mun1z) [reply ml_git/config.py]: Done on last commit.

**Review by lucasncabral — CHANGES_REQUESTED** [review]
Please fix the warning pointed out by flake8:

```
./tests/integration/test_19_create.py:300:107: W292 no newline at end of file
                      + ' --categories=img --mutability=' + STRICT + ' --bucket-name=test --entity-dir='))                                                                                                          ^
1     W292 no newline at end of file
1
```

Also, an integration test is failing:
```
================================= FAILURES =================================
__________ CreateAcceptanceTests.test_12_create_with_unzip_option __________

```

And update the branch to resolve the conflicts.

File: ml_git/config.py [hunk 5]
```diff
@@ -306,6 +306,7 @@ def create_workspace_tree_structure(repo_type, artifact_name, categories, storag
...
     # get root path to create directories and files
     path = get_root_path()
+
```
Comment (GustavoMaciel) [inline ml_git/config.py]: Remove file alteration or update Copyright header

**Review by GustavoMaciel — APPROVED** [review]

**Review by lucasncabral — APPROVED** [review]

**Review by andersonpablosilva — APPROVED** [review]

**Pull request merged**